build/build.go

```diff
@@ -81,14 +85,21 @@ func importWithSrcDir(path string, srcDir string, mode build.ImportMode, install
 			buildContext.InstallSuffix += "_" + installSuffix
 		}
 	}
+
+	// if current path is for gopherjs import and a vendored path for gopherjs has previously been encountered, update
+	// the path to be the vendored one
+	if strings.HasPrefix(path, "github.com/gopherjs/gopherjs") && vendoredGopherJS != "" {
+		path = vendoredGopherJS + strings.TrimPrefix(path, "github.com/gopherjs/gopherjs")
+	}
+
 	pkg, err := buildContext.Import(path, srcDir, mode)
 	if err != nil {
 		return nil, err
 	}
 
-	// TODO: Resolve issue #415 and remove this temporary workaround.
+	// if gopherjs package path is vendored, store it so that we can update future paths
 	if strings.HasSuffix(pkg.ImportPath, "/vendor/github.com/gopherjs/gopherjs/js") {
-		return nil, fmt.Errorf("vendoring github.com/gopherjs/gopherjs/js package is not supported, see https://github.com/gopherjs/gopherjs/issues/415")
+		vendoredGopherJS = strings.TrimSuffix(pkg.ImportPath, "/js")
```

Comment:
How can we be sure that the order of execution will be right? That it will set `vendoredGopherJS` before its value needs to be used above?